pkg/server/endpoints/registration/handler.go

```diff
@@ -229,3 +236,26 @@ func (h *Handler) FetchBundle(
 
 	return &registration.Bundle{CaCerts: b.CaCerts}, nil
 }
+
+func (h *Handler) checkEntryUniqueness(ds datastore.DataStore, entry *common.RegistrationEntry) error {
```

Comment:
nit: this should probably return bool in addition to error, rather than the "i already exist" case being an error itself. doing this will help the error handling nits above too

Reply:
In that case I would change the function name to isEntryUnique(), and the logic would be:
```go	
	unique, err = h.isEntryUnique(dataStore, request)
	if err != nil {
		h.Log.Error(err)
		return nil, errors.New("error trying to create entry")
	}

	if !unique {
		err = errors.New("entry already exists")
		h.Log.Error(err)
		return nil, err		
	}
```
Something like this is what you mean?

Reply:
yea exactly... i realize it probably doesn't make a huge difference in this specific case, but in general the consumer should be able to decide what to do... for instance, maybe in the future we will want to retry this lookup against the datastore if it failed, but not retry if the "problem" is that it's not unique

Reply:
Fair enough, I'll update it.